src/otx/algorithms/common/adapters/mmcv/utils/_builder_build_data_parallel.py

```diff
@@ -134,3 +137,49 @@ def train_step(self, *inputs, **kwargs):
     def val_step(self, *inputs, **kwargs):
         with torch.autocast(device_type="xpu", dtype=torch.bfloat16, enabled=self.enable_autocast):
             return super().val_step(*inputs, **kwargs)
+
+
+class HPUDataParallel(MMDataParallel):
+    def __init__(self, *args, enable_autocast: bool = False, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.enable_autocast = enable_autocast
+        self.src_device_obj = torch.device("hpu", self.device_ids[0])
+
+    def scatter(self, inputs, kwargs, device_ids):
+        inputs, kwargs = super().scatter(inputs, kwargs, [-1])
+
+        for x in inputs:
+            if isinstance(x, tuple):
+                for val in x:
+                    if isinstance(val, dict):
+                        for k in val:
+                            if isinstance(val[k], torch.Tensor):
+                                val[k] = val[k].to(torch.device(f"hpu:{device_ids[0]}"))
```

Comment:
Why don't use `self.src_device_obj` here and in other .to operations?

Reply:
Fixed.